src/agent0/chainsync/exec/acquire_data.py

```diff
@@ -120,66 +109,54 @@ def acquire_data(
-    # monitor for new blocks & add pool info per block
-    if not suppress_logs:
-        logging.info("Monitoring for pool info updates...")
-    while True:
-        latest_mined_block = interfaces[0].web3.eth.get_block_number()
-        # Only execute if we are on a new block
-        if latest_mined_block < curr_write_block:
-            exit_callable = False
-            if exit_callback_fn is not None:
-                exit_callable = exit_callback_fn()
-            if exit_on_catch_up or exit_callable:
-                break
-            time.sleep(_SLEEP_AMOUNT)
-            continue
-        # Backfilling for blocks that need updating
-        if backfill:
-            for block_int in tqdm(range(curr_write_block, latest_mined_block + 1), disable=not progress_bar):
-                block_number: BlockNumber = BlockNumber(block_int)
-                # Only print every 10 blocks
-                if not suppress_logs and (block_number % 10) == 0:
-                    logging.info("Block %s", block_number)
-                # Explicit check against loopback block limit
-                if (latest_mined_block - block_number) > lookback_block_limit:
-                    # NOTE when this case happens, wallet information will no longer
-                    # be accurate, as we may have missed deltas on wallets
-                    # based on the blocks we skipped
-                    # TODO should directly query the chain for open positions
-                    # in this case
-                    logging.warning(
-                        "Querying block_number %s out of %s, unable to keep up with chain block iteration",
-                        block_number,
-                        latest_mined_block,
-                    )
-                    continue
-                data_chain_to_db(interfaces, block_number, db_session)
-        else:
-            data_chain_to_db(interfaces, latest_mined_block, db_session)
-
-        curr_write_block = latest_mined_block + 1
+    # Add all trade events to the table
+    # TODO there may be time and memory concerns here if we're spinning up from
+    # scratch and there's lots of trades/pools.
+    trade_events_to_db(interfaces, wallet_addr=None, db_session=db_session)
+
+    # Add all checkpoint events to the table
+    checkpoint_events_to_db(interfaces, db_session=db_session)
+
+    # Backfilling for blocks that need updating
+    # Note `data_chain_to_db` takes care of handling duplicate rows
+    if backfill:
+        for block_int in tqdm(
+            range(start_block, latest_mined_block + 1, backfill_sample_period),
+            disable=not backfill_progress_bar,
+        ):
+            block_number: BlockNumber = BlockNumber(block_int)
+            # Explicit check against loopback block limit
+            if lookback_block_limit is not None and (latest_mined_block - block_number) > lookback_block_limit:
+                # NOTE when this case happens, wallet information will no longer
```

Comment:
I think it would be a good idea to mention this consequence in the warning message.